### ISSUE TITLE
fix: setting `no-unused-keys` to `error` breaks `next`

### ISSUE THREAD

**Issue by dargmuesli** [intlify/eslint-plugin-vue-i18n#496]
**Tell us about your environment**

- **ESLint version:** 9.0.0
- **eslint-plugin-vue version:** 9.24.1
- **eslint-plugin-vue-i18n version:** 3.0.0-next.9
- **Node version:** 20.11.1

**Please show your full configuration:**

```js
export default [
  ...vueI18n.configs['flat/recommended'],
  {
    rules: {
      '@intlify/vue-i18n/no-unused-keys': 'error', // for all other rules this works fine!
    },
]
```

**What did you do?**
Run eslint

**What did you expect to happen?**
No error to be reported

**What actually happened?**
```
Oops! Something went wrong! :(

ESLint: 9.0.0

ESLint couldn't find a configuration file. To set up a configuration file for this project, please run:

    npm init @eslint/config

ESLint looked for configuration files in /[...project]/src and its ancestors. If it found none, it then looked in your home directory.

If you think you already have a configuration file or if you need more help, please stop by the ESLint Discord server: https://eslint.org/chat

 ELIFECYCLE  Command failed with exit code 2.
```

**Repository to reproduce this issue**

See second commit in https://github.com/dargmuesli/vio/pull/173, adding the rule configuration ([CI run](https://github.com/dargmuesli/vio/actions/runs/8656034042/job/23735935695#step:7:421))

**Comment by dargmuesli** [comment]
Thank you for the quick fix! :heart: Great to see all of you on here again :wink: greetings!
